tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -434,6 +434,145 @@ object SQLPlanParser extends Logging {
     parsedExpressions.distinct.toArray
   }
 
+   // This parser is used for BroadcastHashJoin, ShuffledHashJoin and SortMergeJoin
+   def parseEquijoinsExpressions(exprStr: String): (Array[String], Boolean) = {
+     // ShuffledHashJoin [name#11, CEIL(DEPT#12)], [name#28, CEIL(DEPT_ID#27)], Inner, BuildLeft
+     // SortMergeJoin [name#11, CEIL(dept#12)], [name#28, CEIL(dept_id#27)], Inner
+     // BroadcastHashJoin [name#11, CEIL(dept#12)], [name#28, CEIL(dept_id#27)], Inner,
+     // BuildRight, false
+     val parsedExpressions = ArrayBuffer[String]()
+     // Get all the join expressions and split it with delimiter :: so that it could be used to
+     // parse function names (if present) later.
+     val joinExprs = equiJoinRegexPattern.findAllMatchIn(exprStr).mkString("::")
+     // Get joinType and buildSide(if applicable)
+     val joinParams = equiJoinRegexPattern.replaceAllIn(
+       exprStr, "").split(",").map(_.trim).filter(_.nonEmpty)
+     val joinType = joinParams(0).trim
+     // SortMergeJoin doesn't have buildSide, assign empty string in that case
+     val buildSide = if (joinParams.size > 1) {
+       joinParams(1).trim
+     } else {
+       ""
+     }
+     // Get individual expressions which is later used to get the function names.
+     val expressions = joinExprs.split("::").map(_.trim).map(
+       _.replaceAll("""^\[+|\]+$""", "")).
+       map(_.split(",")).flatten.map(_.trim)
+
+     if (expressions.nonEmpty) {
+       expressions.foreach { expr =>
+         val functionName = getFunctionName(functionPattern, expr)
+         functionName.foreach(parsedExpressions += _)
+       }
+     }
+
+     (parsedExpressions.distinct.toArray, equiJoinSupportedTypes(buildSide, joinType))
+   }
+
+  def parseNestedLoopJoinExpressions(exprStr: String): (Array[String], Boolean) = {
+    // BuildRight, LeftOuter, ((CEIL(cast(id1#1490 as double)) <= cast(id2#1496 as bigint))
+    // AND (cast(id1#1490 as bigint) < CEIL(cast(id2#1496 as double))))
+    val parsedExpressions = ArrayBuffer[String]()
+    // Get joinType and buildSide by splitting the input string.
+    val nestedLoopParameters = exprStr.split(",", 3)
+    val buildSide = nestedLoopParameters(0).trim
+    val joinType = nestedLoopParameters(1).trim
+
+    if (nestedLoopParameters.size > 2) { // condition present on join columns
+      val exprSepAND = if (exprStr.contains("AND")) {
```

Comment:
much of this code look the same as the filterExpresions parser, wondering if we can create some common functions between those?

Reply:
created common function for BroadcastNestedLoopJoinExec and FilterExec